src/common/concurrent/housekeeper.rs

```diff
@@ -1,141 +1,52 @@
 
-// #[cfg(feature = "sync")]
-// use super::constants::{READ_LOG_FLUSH_POINT, WRITE_LOG_FLUSH_POINT};
-
-use parking_lot::Mutex;
-use scheduled_thread_pool::JobHandle;
 use std::{
-    marker::PhantomData,
-    sync::{
-        atomic::{AtomicBool, Ordering},
-        Arc, Weak,
-    },
+    sync::atomic::{AtomicBool, Ordering},
     time::Duration,
 };
 
 pub(crate) trait InnerSync {
-    fn sync(&self, max_sync_repeats: usize) -> Option<SyncPace>;
-
-    // #[cfg(feature = "sync")]
-    // fn now(&self) -> Instant;
+    fn sync(&self, max_sync_repeats: usize);
+    fn now(&self) -> Instant;
 }
 
-#[derive(Clone, Debug)]
-pub(crate) struct Configuration {
-    is_blocking: bool,
-    periodical_sync_enabled: bool,
+pub(crate) struct Housekeeper {
+    is_sync_running: AtomicBool,
+    sync_after: AtomicInstant,
 }
 
-impl Configuration {
-    pub(crate) fn new_thread_pool(periodical_sync_enabled: bool) -> Self {
+impl Default for Housekeeper {
+    fn default() -> Self {
         Self {
-            is_blocking: false,
-            periodical_sync_enabled,
+            is_sync_running: Default::default(),
+            sync_after: AtomicInstant::new(Self::sync_after(Instant::now())),
         }
     }
 }
 
-pub(crate) enum Housekeeper<T> {
-    Blocking(BlockingHousekeeper),
-    ThreadPool(ThreadPoolHousekeeper<T>),
-}
-
-impl<T> Housekeeper<T>
-where
-    T: InnerSync + 'static,
-{
-    pub(crate) fn new(inner: Weak<T>, config: Configuration) -> Self {
-        if config.is_blocking {
-            Housekeeper::Blocking(BlockingHousekeeper::default())
-        } else {
-            Housekeeper::ThreadPool(ThreadPoolHousekeeper::new(
-                inner,
-                config.periodical_sync_enabled,
-            ))
-        }
+impl Housekeeper {
+    pub(crate) fn should_apply_reads(&self, ch_len: usize, now: Instant) -> bool {
+        self.should_apply(ch_len, READ_LOG_FLUSH_POINT / 8, now)
     }
 
-    // #[cfg(feature = "sync")]
-    // pub(crate) fn should_apply_reads(&self, ch_len: usize, now: Instant) -> bool {
-    //     match self {
-    //         Housekeeper::Blocking(h) => h.should_apply_reads(ch_len, now),
-    //         Housekeeper::ThreadPool(h) => h.should_apply_reads(ch_len, now),
-    //     }
-    // }
-
-    // #[cfg(feature = "sync")]
-    // pub(crate) fn should_apply_writes(&self, ch_len: usize, now: Instant) -> bool {
-    //     match self {
-    //         Housekeeper::Blocking(h) => h.should_apply_writes(ch_len, now),
-    //         Housekeeper::ThreadPool(h) => h.should_apply_writes(ch_len, now),
-    //     }
-    // }
-
-    pub(crate) fn try_sync(&self, cache: &impl InnerSync) -> bool {
-        match self {
-            Housekeeper::Blocking(h) => h.try_sync(cache),
-            Housekeeper::ThreadPool(h) => h.try_schedule_sync(),
-        }
+    pub(crate) fn should_apply_writes(&self, ch_len: usize, now: Instant) -> bool {
+        self.should_apply(ch_len, WRITE_LOG_FLUSH_POINT / 8, now)
```

Comment:
Rather than dividing by 8 here, change the value of the `READ_` and `WRITE_LOG_FLUSH_POINT`s.